src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs

```diff
@@ -113,7 +114,7 @@ public void FlushCachedData()
             let containsExecutable = image.Any(entry => entry.IsExecutable)
             let beginAddress = image.Min(entry => entry.BeginAddress)
             let props = GetPEImageProperties(filePath)
-            select new ModuleInfo(beginAddress, props.Filesize, props.Timestamp, filePath, containsExecutable, buildId: default, version: props.Version);
+            select new ModuleInfo(beginAddress, props.Filesize, props.Timestamp, filePath, containsExecutable, buildId: ImmutableArray<byte>.Empty, version: props.Version);
```

Comment:
Is build ID unavailable in live processes?

Reply:
That original code is wrong.  It exists but it isn't needed by ClrMD itself.  In the live scenario we should almost always have libmscordaccore.so on disk (unless the user published a build and/or deleted the file from disk) so we wouldn't need to hit the symbol server for it.  (I also realize ClrMD hasn't added symbol server support for buildid based files.)

In this case though, I think removing the parameter from the constructor and always lazily pulling the value is the right approach.